### PR TITLE
add docs for verify-webhook action

### PR DESCRIPTION
**Pull request by Megalonia**

### CODE REVIEW

File: docs/http/traffic-policy/actions/verify-webhooks.mdx
Comment (Megalonia) [inline docs/http/traffic-policy/actions/verify-webhooks.mdx:1]: maybe something with all the integration would be nice

File: docs/http/traffic-policy/actions/index.mdx [hunk 1]
```diff
@@ -23,3 +23,4 @@ Traffic Policy actions enable you to modify the behavior of traffic flowing thro
 | [remove-headers](remove-headers)       | Remove headers from incoming requests or outgoing responses.          | on_http_request, on_http_response                 |
 | [restrict-ips](restrict-ips)           | Allow or deny traffic based on source IP.                             | on_http_request, on_tcp_connect                   |
 | [url-rewrite](url-rewrite)             | Rewrite request URLs transparently using regular expressions.         | on_http_request                                   |
+| [verify-webhook](verify-webhook)       | Ensures webhook requests sent to your endpoints are intended for you. | on_http_request                                   |
```
Comment (nijikokun) [inline docs/http/traffic-policy/actions/index.mdx:26]:
```suggestion
| [verify-webhook](verify-webhook)       | Validate incoming webhook signatures against a known secret to ensure authenticity. | on_http_request                                   |
```

File: traffic-policy/actions/verify-webhook/index.mdx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+## Overview
+
+The **Verify Webhook** Traffic Policy action enables you to validate incoming webhook signatures against a known secret to ensure authenticity, and either forwards the request to the next action or rejects it based on the verification result.
```
Comment (nijikokun) [inline traffic-policy/actions/verify-webhook/index.mdx:3]:
```suggestion
The **Verify Webhook** Traffic Policy action enables you to validate incoming webhook signatures against a known secret to ensure authenticity. Depending on the verification result, it either forwards the request to the next action or rejects it, safeguarding your endpoints from unauthorized or tampered webhooks.
```

File: traffic-policy/actions/verify-webhook/config.mdx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+`verify-webhook`
+
+### Configuration Fields
+
+| Parameter  | Type     | Description                                                                                                           |
+| ---------- | -------- | --------------------------------------------------------------------------------------------------------------------- |
+| `provider` | `string` | **Required.** The name of the provider to verify webhook requests from. Value must be one of the supported providers. |
+| `secret`   | `string` | A secret key used to validate requests from the given provider.                                                       |
+| `enforce`  | `bool`   | Whether or not the actions is executed (defaults to true)`.                                                           |
```
Comment (nijikokun) [inline traffic-policy/actions/verify-webhook/config.mdx:16]:
```suggestion
| `enforce`  | `bool`   | Default `true`. If `false`, continue to the next action even if the request is not allowed.                                                          |
```

File: traffic-policy/actions/verify-webhook/config.mdx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+### Action Type
+
+`verify-webhook`
+
+### Configuration Fields
+
+| Parameter  | Type     | Description                                                                                                           |
+| ---------- | -------- | --------------------------------------------------------------------------------------------------------------------- |
+| `provider` | `string` | **Required.** The name of the provider to verify webhook requests from. Value must be one of the supported providers. |
+| `secret`   | `string` | A secret key used to validate requests from the given provider.                                                       |
```
Comment (nijikokun) [inline traffic-policy/actions/verify-webhook/config.mdx:15]:
```suggestion
| `secret`   | `string` | The secret key used to validate webhook requests from the specified provider.                                                   |
```

File: traffic-policy/actions/verify-webhook/config.mdx [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+### Action Type
+
+`verify-webhook`
+
+### Configuration Fields
+
+| Parameter  | Type     | Description                                                                                                           |
+| ---------- | -------- | --------------------------------------------------------------------------------------------------------------------- |
+| `provider` | `string` | **Required.** The name of the provider to verify webhook requests from. Value must be one of the supported providers. |
```
Comment (nijikokun) [inline traffic-policy/actions/verify-webhook/config.mdx:14]:
```suggestion
| `provider` | `string` | **Required.** The name of the provider to verify webhook requests from. Value must match a supported provider. |
```

File: traffic-policy/actions/verify-webhook/behavior.mdx [hunk 6]
```diff
@@ -0,0 +1,27 @@
+### Behavior
+
+The **verify-webhook** action validates that incoming webhook requests are authentic. When a request is received, the action verifies the request by validating its signature against a known secret. If the verification is successful, the action allows the request to continue through the action chain and to finally to your application; if verification fails, the request will be terminated with a `403 Forbidden` response.
```
Comment (nijikokun) [inline traffic-policy/actions/verify-webhook/behavior.mdx:3]:
```suggestion
The **verify-webhook** action ensures the authenticity of incoming webhook requests by validating their signatures against a known secret. Upon receiving a request, the action performs the signature verification. If verification succeeds, the request proceeds through the action chain. If it fails, the request is terminated with a `403 Forbidden` response, unless `enforce` is set to `false`, in which case the request proceeds without termination.
```

File: traffic-policy/actions/verify-webhook/behavior.mdx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+#### Verification Process
+
+- **Signature Validation**: The action validates incoming webhook signature to confirm the request originates from the configured provider and that the payload has not been tampered with.
+- **Request Handling**: If the webhook verification is successful, the request is forwarded to the next action. If the verification fails, the request chain is terminated with a `403` response.
+- **Configurable Enforcement**: By default, verification failures result in termination. However, setting `enforce: false` allows unverified requests to proceed, while logging the verification result. This option is for debugging and testing.
```
Comment (nijikokun) [inline traffic-policy/actions/verify-webhook/behavior.mdx:9]:
```suggestion
- **Configurable Enforcement**: By default, verification failures result in termination. However, setting `enforce: false` allows unverified requests to proceed, while logging the verification result. This option is useful for debugging, testing, and crafting your own custom responses with action result variables.
```

File: traffic-policy/actions/verify-webhook/examples/basic-example.mdx [hunk 8]
```diff
@@ -0,0 +1,62 @@
+import ConfigExample from "/src/components/ConfigExample.tsx";
+
+### Basic Example
+
+This example configuration sets up an endpoint (`gitlab-webhook-example.ngrok.io`) that receives webhook requests from GitLab. The **Verify Webhook** action checks if the request is authentic based on a shared secret. If verified, a custom response is sent back with a 200 status and a plain text confirmation message.
```
Comment (nijikokun) [inline traffic-policy/actions/verify-webhook/examples/basic-example.mdx:5]:
```suggestion
This example configuration sets up an endpoint (`gitlab-webhook-example.ngrok.io`) that receives webhook requests from GitLab. The **Verify Webhook** action checks the authenticity of the request using a shared secret. If the request is verified, a custom response is sent back with a status `200 OK` and a plain text confirmation message.
```

File: traffic-policy/actions/verify-webhook/examples/basic-example.mdx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+--header 'X-Gitlab-Token: secret!'
+> POST / HTTP/2
+> Host: gitlab-webhook-example.ngrok.io
+> User-Agent: curl/[version]
+> Accept: */*
+> X-Gitlab-Token: secret!
+...
+```
+
+This request will first be processed by the Verify Webhook action. If the GitLab webhook verification is successful, ngrok will return a 200 OK response with the message GitLab webhook verified.
```
Comment (nijikokun) [inline traffic-policy/actions/verify-webhook/examples/basic-example.mdx:56]:
```suggestion
This request will first be processed by the Verify Webhook action. If the GitLab webhook verification is successful, ngrok will return a `200 OK` response with the message GitLab webhook verified.
```

File: traffic-policy/actions/verify-webhook/providers.mdx [hunk 10]
```diff
@@ -0,0 +1,71 @@
+### Supported Providers {#supported-providers}
```
Comment (nijikokun) [inline traffic-policy/actions/verify-webhook/providers.mdx:1]: We should go through and update all of these to show traffic policy, for now we should probably show a note that the integration guides may refer to the module.

**Review by DevMandy — APPROVED** [review]
lgtm

**Pull request merged**